network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -500,22 +602,31 @@ func (c *ControlMsgValidationInspector) inspectRpcPublishMessages(from peer.ID,
 		// cluster prefix status is unnecessary when the error threshold is exceeded.
 		err, _ := c.validateTopic(from, topic, activeClusterIDS)
 		if err != nil {
-			// we can skip checking for subscription of topic that failed validation and continue
+			// short circuit validation when unstaked peer detected
+			if IsErrUnstakedPeer(err) {
+				lg.Debug().Msg("received control message from unstaked peer")
+				continue
```

Comment:
⚠️ it seems this part needs reconsideration; if we keep _continuing_ the for loop, we never report the errors for unstated peers (on cluster prefix channels). Is it the desirable behavior?

Reply:
https://github.com/onflow/flow-go/pull/4978/commits/8a28bf97bad6e97fb921b27887f94b315c26d5f4